### PR TITLE
KEYCLOAK-2743: Port OAuthClient to new testsuite

### PR DESCRIPTION
**Pull request by ssilvert**
This also includes a port of most of the admin tests, several of which used the migrated OAuthClient.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,6 +63,7 @@
         <jboss.logging.tools.wf8.version>1.2.0.Final</jboss.logging.tools.wf8.version>
         <jboss-jaxrs-api_2.0_spec>1.0.0.Final</jboss-jaxrs-api_2.0_spec>
         <jboss.spec.javax.xml.bind.jboss-jaxb-api_2.2_spec.version>1.0.4.Final</jboss.spec.javax.xml.bind.jboss-jaxb-api_2.2_spec.version>
+        <org.json.version>20080701</org.json.version>
```
Comment (stianst) [inline pom.xml]: Please don't use org.json we use fasterxml/jackson and don't want to json libraries

**Comment by stianst** [comment]
There's a lot of changes here that are not related to porting OAuthClient. MigratedOAuthClient is not a nice name. Merge it org.keycloak.testsuite.util.OAuthClient instead of adding two separate, this will also make it ready to inject into tests.

**Comment by ssilvert** [comment]
I was under the impression that this needs to be done right away and we could refactor later.  But I can spend time on trying to unify the two OAuthClients now if that's what you want.

**Comment by abstractj** [comment]
@ssilvert I got confused about the migrated.admin package. Wouldn't make more sense to just remove "migrated" from this and merge it under the admin package?

**Comment by ssilvert** [comment]
Those classes under org.keycloak.admin.<something> are the new REST API tests.  So I didn't want to confuse things by putting migrated tests under org.keycloak.admin.  They definitely don't go there as long as our REST API tests are where they are.

We probably do need to rethink the whole package naming scheme in this test suite.  For now, I think there is value in knowing which classes were migrated from the old testsuite.  So having "migrated" in the package path is, IMO, a good thing for the time being.

**Pull request closed without merging**